src/lib/storage/segment_iterables/any_segment_iterable.hpp

```diff
@@ -75,8 +75,9 @@ class AnySegmentIterableWrapper : public BaseAnySegmentIterableWrapper<ValueType
   void with_iterators(const std::shared_ptr<const AbstractPosList>& position_filter,
                       const AnySegmentIterableFunctorWrapper<ValueType>& functor_wrapper) const override {
     if (position_filter) {
-      if constexpr (is_point_accessible_segment_iterable_v<IterableT>) {
-        iterable.with_iterators(position_filter, [&](auto begin, const auto end) {
+      if constexpr (is_point_accessible_segment_iterable_v<UnerasedIterable>) {
+        // Since the iterator type is erased, there is no reason to resolve the PosList
+        iterable.template with_iterators<ErasePosListType::Always>(position_filter, [&](auto begin, const auto end) {
```

Comment:
The comment in line 79 needs to be a bit more clear. I assume it means something like

> We delete the iterator type and deliberately sacrifice performance here. Thus, we don't need to resolve the PosList type, since we are already ignoring performance. (instead, we focus on compile time)

Reply:
Yepp, I can see where the confusion comes from. This is entirely independent of the if statement. Is this better?

> Since we are in AnySegmentIterable, where we erase the segment types as far as possible, there is no reason to resolve the PosList. This further reduces the compile time at the cost of run time performance (which we have already sacrificed when we chose the AnySegmentIterable in the first place).